BrainPortal/lib/globus_helpers.rb

```diff
@@ -75,161 +64,143 @@ def globus_fetch_token(code, globus_action_url)
-  # globus negotiations. The current Rails session_id, encoded, will do
+  # OpenID negotiations. The current Rails session_id, encoded, will do
   # the trick. The Rails session is maintained by a cookie already
   # created and maintained, at this point.
-  def globus_current_state
-    Digest::MD5.hexdigest( request.session_options[:id] )
-  end
-
-  # Return the registered globus endpoint client ID.
-  # This value must be configured by the CBRAIN admin
-  # in the meta data of the portal. Returns nil if unset.
-  def globus_client_id
-    myself = RemoteResource.current_resource
-    myself.meta[:globus_client_id].presence.try(:strip)
-  end
-
-  # Return the registered globus endpoint client secret.
-  # This value must be configured by the CBRAIN admin
-  # in the meta data of the portal. Returns nil if unset.
-  def globus_client_secret
-    myself = RemoteResource.current_resource
-    myself.meta[:globus_client_secret].presence.try(:strip)
+  def oidc_current_state(oidc)
+    Digest::MD5.hexdigest( request.session_options[:id] ) + "_" + oidc.name
   end
 
   # Returns true if the CBRAIN system is configured for
-  # globus auth.
-  def globus_auth_configured?
+  # OIDC auth.
+  def oidc_auth_configured?(oidc)
     myself   = RemoteResource.current_resource
     site_uri = myself.site_url_prefix.presence
-    # Three conditions: site uri, client ID, client secret.
+
     return false if ! site_uri
-    return false if ! globus_client_id
-    return false if ! globus_client_secret
+    return false if ! oidc.client_id
+    return false if ! oidc.client_secret
+    return false if ! oidc.authorize_uri
+    return false if ! oidc.scope
     true
   end
 
-  # Record the globus identity for the current user.
+  # Record the OIDC identity for the current user.
   # (This maybe should be made into a User model method)
-  def record_globus_identity(user, globus_identity)
-
+  def record_oidc_identity(oidc, user, oidc_identity) #:nodoc:
     # In the case where a user must auth with a specific set of
-    # globus providers, we find the first identity that
+    # OIDC providers, we find the first identity that
     # matches a name of that set.
-    identity = set_of_identities(globus_identity).detect do |idstruct|
-       user_can_link_to_globus_identity?(user, idstruct)
+
+    identity = set_of_identities(oidc_identity).detect do |idstruct|
+        user_can_link_to_oidc_identity?(oidc, user, idstruct)
     end
 
-    provider_id   = identity['identity_provider']              || cb_error("Globus: No identity provider")
-    provider_name = identity['identity_provider_display_name'] || cb_error("Globus: No identity provider name")
-    pref_username = identity['preferred_username'] ||
-                    identity['username']                       || cb_error("Globus: No preferred username")
+    provider_id   = identity[oidc.identity_provider_key]              || cb_error("#{oidc.name}: No identity provider")
```

Comment:
Use `oidc.identity_info()` instead. That is what this method is for. You should try to hide any identity struct accesses by using a method, to make the code cleaner and avoid bugs (like the next comment I will leave about method `find_user_with_globus_identity`)